File: 백준/Silver/30987. 하루 피부과/하루 피부과.py
import sys
input = sys.stdin.readline


def main():
    # 입력값
    x1, x2 = map(int, input().split())
    a, b, c, d, e = map(int, input().split())
    
    # 풀이
    def integral(x):
        return (a / 3) * x**3 + (b - d) / 2 * x**2 + (c - e) * x

    result = int(integral(x2) - integral(x1))
    # 출력
    print(result)


if __name__ == "__main__":
    main()

File: 백준/Silver/30987. 하루 피부과/하루 피부과.cc
#include <iostream>
using namespace std;

double integral(int x, int a, int b, int c, int d, int e) {
    return (a / 3.0) * x * x * x + (b - d) / 2.0 * x * x + (c - e) * x;
}

int main() {
    // 입력값
    int x1, x2;
    cin >> x1 >> x2;

    int a, b, c, d, e;
    cin >> a >> b >> c >> d >> e;

    // 풀이
    double result = integral(x2, a, b, c, d, e) - integral(x1, a, b, c, d, e);

    // 출력
    cout << static_cast<int>(result) << endl;

    return 0;
}
